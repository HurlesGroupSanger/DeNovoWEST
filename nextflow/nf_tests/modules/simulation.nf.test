nextflow_process {

    name "Simulation module"
    script "modules/simulation.nf"
    process "SIMULATION"

    test("simulation_OR4F5") {

        when {

            params {
                dnm = "$baseDir/nf_tests/test_data/simulation/CADD/input/OR4F5.dnms.annotated.tsv"
                rates = "$baseDir/nf_tests/test_data/simulation/CADD/input/OR4F5.rates.annotated.tsv.gz"
                score="score"
                runtype = "ns"

                nmales = 17422
                nfemales = 13636
            }
            process {
                """
                input[0] = Channel.fromPath(params.dnm).combine(Channel.fromPath(params.rates))
                input[1] = params.score
                input[2] = params.nmales
                input[3] = params.nfemales
                input[4] = params.runtype

                """
            }
        }

        then {

            assert process.success

            // Test that the output matches the latest output
            //assert snapshot(process.out).match()
            
            // Check that the output example is still up to date
            //assert path(process.out.get(0).get(0)).md5 == path("$baseDir/../examples/simulation/CADD/output/OR4F5.enrichment_results.tsv").md5
        }
    }
}
