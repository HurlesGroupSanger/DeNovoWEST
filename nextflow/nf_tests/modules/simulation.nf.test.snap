{
    "simulation_OR4F5": {
        "content": [
            {
                "0": [
                    "enrichment_results.tsv:md5,dd866cc1ad7a7831bd03b7453bb01767"
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "dnm": [
                    
                ],
                "rates": [
                    
                ],
                "results": [
                    "enrichment_results.tsv:md5,dd866cc1ad7a7831bd03b7453bb01767"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-01-13T19:03:12.518765097"
    }
}